pkg/cmd/waitforkube/waitforkube.go

```diff
@@ -34,31 +37,53 @@ func NewWaitForKubeCommand(errOut io.Writer) *cobra.Command {
 				if err := fn(); err != nil {
 					if cmd.HasParent() {
 						klog.Fatal(err)
-						fmt.Fprint(waitForKubeOpts.errOut, err.Error())
+						fmt.Fprint(waitForKube.errOut, err.Error())
 					}
 				}
 			}
-			must(waitForKubeOpts.Run)
+			must(waitForKube.Run)
 		},
 	}
 
 	return cmd
 }
 
-func (w *waitForKubeOpts) Run() error {
+func (w *waitForKube) Run() error {
+	dataFilePath := filepath.Join(dataDir, dbPath)
+	if CheckEtcdDataFileExists(dataFilePath) {
+		klog.Infof("etcd has previously been initialized as a member")
+		return nil
+	}
 	wait.PollInfinite(retryDuration, func() (bool, error) {
 		if _, err := os.Stat(saTokenPath); os.IsNotExist(err) {
 			klog.Infof("waiting for kube service account resources to sync: %v", err)
 			return false, nil
 		}
+		klog.Infof("kube dependencies verified, service account resources found")
 		return true, nil
 	})
 	if !inCluster() {
-		return fmt.Errorf("kube env not populated")
+		return fmt.Errorf("container missing required env KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT")
```

Comment:
if this is running as a static pods these envs are not present and this was blocking the pod startup as it is an init container

Reply:
When kube comes up and the container restarts on this error these are then available. I have been using this as a quick sanity check. But I am open to other ideas.

Reply:
If it didn't work we would never scale :)

Reply:
An example where these are valid without a restart.

https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_installer/3051/pull-ci-openshift-installer-master-e2e-aws/9952/artifacts/e2e-aws/pods/openshift-etcd_etcd-member-ip-10-0-130-38.us-east-2.compute.internal_wait-for-kube.log

Reply:
But yes in recovery or lights out this will never happen that's is why we need the changes in this PR